src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java

```diff
@@ -66,7 +66,12 @@ public static void addRunFacet(@Nonnull ContainerRecord record, @Nonnull Run<?,?
     /**
      * Creates a new {@link DockerFromFingerprintFacet} and adds a run. Or adds to an existing one.
      */
-    public static void addFromFacet(@Nonnull String imageId, @Nonnull Run<?,?> run) throws IOException {
+    public static void addFromFacet(@Nonnull String baseImageId, @Nonnull String derivedImageId, @Nonnull Run<?,?> run) throws IOException {
```

Comment:
Agreed that using the term “base image” is misleading in this context. The problem is not helped by the fact that Docker documentation [also uses this term in relation to the `FROM` instruction](https://docs.docker.com/reference/builder/#from).

“Parent image” is not exactly right for the first parameter here, though. It could in fact be the great-great-grandparent:

```
FROM java:openjdk-7u65-jdk
COPY something /tmp
COPY something-else /tmp
COPY yet-again /tmp
COPY still-more /tmp
```

Here the second parameter should represent the image with `/tmp/still-more`. The parent lacks this but has `/tmp/yet-again`. What we want for the first parameter is the ID of `java:openjdk-7u65-jdk` (not a base image in this case, but could be if we chose for example `ubuntu`—does not matter).

So perhaps it should just be called `fromImageId`. I suppose it could be permitted to be null, in the rare case you were actually creating a base image from a Jenkins job, using

```
FROM scratch
```